sub-packages/bionemo-scdl/src/bionemo/scdl/index/row_feature_index.py

```diff
@@ -160,12 +173,14 @@ def number_of_values(self) -> List[int]:
             self._cumulative_sum_index[i] - max(self._cumulative_sum_index[i - 1], 0)
             for i in range(1, len(self._cumulative_sum_index))
         ]
-
-        vals = [n_rows * len(self._feature_arr[i]) for i, n_rows in enumerate(rows)]
+        vals = []
+        for i, n_rows in enumerate(rows):
+            num_genes = self._num_genes_per_row[i]
+            vals.append(n_rows * num_genes)
```

Comment:
vals = [n_rows * len(self._num_genes_per_row[i]) for i, n_rows in enumerate(rows)]